src/engine/strat_engine/thindev.rs

```diff
@@ -54,27 +66,75 @@ impl ThinDev {
 
     }
 
-    /// Use DM to create a thin provisioned DM device.
-    pub fn setup(&mut self,
-                 dm: &DM,
-                 thin_pool: &mut ThinPoolDev,
-                 thin_id: u32,
-                 length: &Sectors)
-                 -> EngineResult<()> {
+    pub fn teardown(&mut self, dm: &DM) -> EngineResult<()> {
+        try!(dm.device_remove(&DevId::Name(&self.name), DmFlags::empty()));
+        Ok(())
+    }
+
+    pub fn path(&self) -> EngineResult<PathBuf> {
+        match self.dev_info.device().path() {
+            Some(path) => return Ok(path),
+            None => {
+                return Err(EngineError::Engine(ErrorEnum::Invalid,
+                                               "No path associated with dev_info".into()))
+            }
+        }
+    }
 
-        try!(thin_pool.message(dm, &format!("create_thin {}", thin_id)));
-        try!(dm.device_create(&self.name, None, DmFlags::empty()));
-        let table = self.dm_table(&thin_pool.dev_info.unwrap().device().dstr(),
-                                  thin_id,
-                                  length);
-        let id = &DevId::Name(&self.name);
-        self.dev_info = Some(try!(dm.table_load(id, &table)));
-        try!(dm.device_suspend(id, DmFlags::empty()));
+    pub fn create_fs(&mut self) -> EngineResult<()> {
+        let path_result = try!(self.path());
```

Comment:
If I change this to:

 let dev_path = Path::new(&try!(self.path()));

Am I misunderstanding your comment?
I get "borrowed value does not live long enough: consider using a `let` binding to increase its lifetime temporary value dropped here while still  borrowed"